Java heap space

My node regulary crashes with error
```
Uncaught error from thread [Uncaught error from thread [ergoref-akka.actor.default-dispatcher-2]: ergoref-scheduler-1]: Java heap spaceUncaught error from thread [ergoref-akka.io.pinned-disp
atcher-10]: Java heap space, shutting down JVM since 'akka.jvm-exit-on-fatal-error' is enabled for ActorSystem[ergoref]
java.lang.OutOfMemoryError: Java heap space
```
I run the node with `-Xmx4G` parameter so there should be enough memory available. 
I guess that the problem is with the new mempool, node is generating transactions `maxTransactionsPerBlock = 200`

Another node crashed with
```
[ERROR] [SECURITY][02/07/2019 01:41:47.846] [ergoref-akka.actor.default-dispatcher-2] [akka.actor.ActorSystemImpl(ergoref)] Uncaught error from thread [ergoref-akka.actor.default-dispatcher-2]: GC overhead limit exceeded, shutting down JVM since 'akka.jvm-exit-on-fatal-error' is enabled for ActorSystem[ergoref]
java.lang.OutOfMemoryError: GC overhead limit exceeded
```

--- a/src/main/scala/org/ergoplatform/nodeView/wallet/ErgoWalletActor.scala
+++ b/src/main/scala/org/ergoplatform/nodeView/wallet/ErgoWalletActor.scala
@@ -149,6 +149,7 @@ class ErgoWalletActor(ergoSettings: ErgoSettings) extends Actor with ScorexLoggi
       }
 
     case ScanOnchain(fullBlock) =>
+      prover.IR.resetContext()
       height = fullBlock.header.height
       fullBlock.transactions.flatMap(tx => scan(tx, Some(height))).foreach { tb =>
         self ! Resolve(Some(tb.boxId))

